extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -25,17 +25,66 @@ export class BaseCarousel extends AMP.BaseElement {
     /** @private {!Element} */
     this.nextButton_;
 
-    this.buildCarousel();
-    this.buildButtons();
-    this.setupGestures();
-    this.setControlsState();
+    /** @private {?number} */
+    this.autoplayTimeoutId_ = null;
```

Comment:
Why are these here? `autoplay` and `loop` concepts are only applicable to slides, right?